### PR TITLE
[VC] support pod scheduling

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### CODE REVIEW

**Comment by zhuangqh** [comment]
/hold
need e2e test

File: incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/pod/controller.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		}
+		if err := c.SchedulerEngine.DeSchedulePod(podKey); err != nil {
+			return reconciler.Result{}, fmt.Errorf("failed to unreserve pod %s in %s: %v", request.Name, request.ClusterName, err)
+		}
+	}
+
+	candidate := internalcache.NewPod(request.ClusterName, pod.Namespace, pod.Name, string(pod.UID), "", util.GetPodRequirements(pod))
+	ret, err := c.SchedulerEngine.SchedulePod(candidate)
+	if err != nil {
+		return reconciler.Result{}, fmt.Errorf("failed to schedule namespace %s in %s: %v", request.Name, request.ClusterName, err)
```
Comment (Fei-Guo) [inline incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/pod/controller.go:122]: schedule namespace -> schedule pod

File: incubator/virtualcluster/experiment/pkg/scheduler/cache/snapshot.go [hunk 2]
```diff
@@ -135,3 +135,58 @@ func (c *schedulerCache) SnapshotForNamespaceSched(nsToRemove ...*Namespace) (*N
...
+	for _, place := range ns.schedule {
+		capability := v1.ResourceList{}
+
+		for k, v := range ns.quotaSlice {
+			val := v.DeepCopy()
+			val.Set(v.Value() * int64(place.num))
+			capability[k] = val
+		}
+		s.clusterUsageMap[place.cluster] = &ClusterUsage{
+			capacity: capability,
```
Comment (Fei-Guo) [inline incubator/virtualcluster/experiment/pkg/scheduler/cache/snapshot.go:187]: Where do track the allocated capacity for each cluster? I suppose you need to traverse the Pod map in each cluster.

File: incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/pod/controller.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	pod := podObj.(*v1.Pod)
+	if c.skipPodSchedule(pod) {
+		// skip irrelevant pod update event
+		// we assume pod's scheduling info won't be manually mutated during pod running by now.
+		return reconciler.Result{}, nil
+	}
+
+	// A Pod delete event followed by an immediate Pod add event may be merged
+	// into a Pod update event. In this case, we should first clean cache.
+	if cPod := c.SchedulerEngine.GetPod(podKey); cPod != nil {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/experiment/pkg/scheduler/resource/virtualcluster/pod/controller.go:108]: 108 -117 can be removed once we remove the uid from the request.

**Pull request merged**